public/app/features/dashboard/state/DashboardModel.ts

```diff
@@ -241,6 +243,26 @@ export class DashboardModel {
     return copy;
   }
 
+  /**
+   * This will load a new dashboard, but keep existing panels unchanged
+   *
+   * This function can be used to implement:
+   * 1. potentially faster loading dashboard loading
+   * 2. dynamic dashboard behavior
+   * 3. "live" dashboard editing
+   *
+   * @internal and experimental
+   */
+  updatePanels(panels: IPanelModel[]): PanelMergeInfo {
+    const info = mergePanels(this.panels, panels ?? []);
+    if (info.changed) {
+      this.panels = info.panels ?? [];
+      this.sortPanelsByGridPos();
+      this.events.publish(new DashboardPanelsChangedEvent());
+    }
+    return info;
+  }
+
```

Comment:
Just spit balling here, instead of adding another public function to the DashboardModel can we dispatch an event? Or maybe we don't have support for that? I'm thinking that could be a better way to communicate from a plugin?

Reply:
yes, I think an event is a good idea, but the general issues is still what API we really want to expose and support long term.  The proposal here is an internal API that is not exposed anywhere.  External plugins would need to use:
```
((...getDashboardSrv().current()) as any)).updatePanels( ...)
```
not pretty!  but also gives us some time to figure out how to better expose.

If by "dispatch" you mean redux -- we currently do not let any plugins touch the redux store.

Reply:
Sorry I was unclear, I didn't mean redux in this context.

Reply:
Until we figure all things out, like how to call this, template variable etc, I think we should put this feature behind a feature toggle so we can get this merged.